hw3/hw3-v1tal4ik/src/v1tal4ik.js

```diff
@@ -0,0 +1,161 @@
+// eslint-disable-next-line max-classes-per-file
+import validateObj from './helperFunc';
+import * as objUsers from './people.data';
+
+// 1. write a constructor function `PersonFunc` whith methods : getFullName() getAge() getFullAddress();
+const PersonFunc = function (obj) {
+  const { isObjHasAllProperty, isObjPropertyTypeRight, msg } = validateObj(obj);
+  if (!isObjHasAllProperty || !isObjPropertyTypeRight) {
+    return { msg };
+  }
+  this.person = obj;
+};
+
+
+PersonFunc.prototype.getFullName = function () { return `${this.person.firstName} ${this.person.lastName}`; };
+PersonFunc.prototype.getFullAddress = function () { return `${this.person.address.country}, ${this.person.address.city}, ${this.person.address.street}, ${this.person.address.house}/${this.person.address.apartment}`; };
+PersonFunc.prototype.getAge = function () {
+  const nowYear = new Date().getFullYear();
+  const birthYear = this.person.birthDate.split('.').pop();
+  return `${nowYear - birthYear} years`;
+};
+
+
+// 2. write a constructor function `WorkingPersonFunc` which inherits from `PersonFunc` and has extra method: getProfessionalNameAndRank();
+const WorkingPersonFunc = function (obj) {
+  const result = PersonFunc.call(this, obj);
+  if (!(result === undefined)) {
+    return result;
+  }
+};
+
+
+WorkingPersonFunc.prototype = Object.create(PersonFunc.prototype);
+WorkingPersonFunc.prototype.constructor = WorkingPersonFunc;
+WorkingPersonFunc.prototype.getProfessionalNameAndRank = function () {
+  const { firstName, lastName, job: { title, experience } } = this.person;
+  return `${firstName} ${lastName}, ${title}, job experiance ${experience} years`;
```

Comment:
for getting `${firstName} ${lastName}` you could use parent's method getFullName()

Reply:
I thought that if I was already using object destruction, it would be shorter and easier.